src/main/notifications/index.ts

```diff
@@ -92,34 +92,48 @@ class NotificationManager {
                 log.debug('notification timeout', serverName, mention.uId);
                 resolve({status: 'error', reason: 'notification_timeout'});
             }, 10000);
+            let failed = false;
 
             mention.on('show', () => {
-                log.debug('displayMention.show', serverName, mention.uId);
-
-                // On Windows, manually dismiss notifications from the same channel and only show the latest one
-                if (process.platform === 'win32') {
-                    const mentionKey = `${mention.teamId}:${mention.channelId}`;
-                    if (this.mentionsPerChannel.has(mentionKey)) {
-                        log.debug(`close ${mentionKey}`);
-                        this.mentionsPerChannel.get(mentionKey)?.close();
-                        this.mentionsPerChannel.delete(mentionKey);
+                // Ensure the failed event isn't also called, if it is we should resolve using its method
+                setTimeout(() => {
+                    if (!failed) {
+                        log.debug('displayMention.show', serverName, mention.uId);
+
+                        // On Windows, manually dismiss notifications from the same channel and only show the latest one
+                        if (process.platform === 'win32') {
+                            const mentionKey = `${mention.teamId}:${mention.channelId}`;
+                            if (this.mentionsPerChannel.has(mentionKey)) {
+                                log.debug(`close ${mentionKey}`);
+                                this.mentionsPerChannel.get(mentionKey)?.close();
+                                this.mentionsPerChannel.delete(mentionKey);
+                            }
+                            this.mentionsPerChannel.set(mentionKey, mention);
+                        }
+                        const notificationSound = mention.getNotificationSound();
+                        if (notificationSound) {
+                            MainWindow.sendToRenderer(PLAY_SOUND, notificationSound);
+                        }
+                        flashFrame(true);
+                        clearTimeout(timeout);
+                        resolve({status: 'success'});
                     }
-                    this.mentionsPerChannel.set(mentionKey, mention);
-                }
-                const notificationSound = mention.getNotificationSound();
-                if (notificationSound) {
-                    MainWindow.sendToRenderer(PLAY_SOUND, notificationSound);
-                }
-                flashFrame(true);
-                clearTimeout(timeout);
-                resolve({status: 'success'});
+                }, 0);
```

Comment:
Is there any effect on adding this timeout if the on `show` is already running asynchronously ? I was expecting some min time set here to wait

Reply:
This is basically there to tell the JS engine to requeue this logic at the end of the event queue so that by then if the `failed` event had fired, the flag would be set and we wouldn't run both callbacks (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Event_loop#zero_delays)

I'm really not happy with this solution, but it's all I could think of to try and make sure we resolved the promise correctly, otherwise it's a race between the two callbacks.

Reply:
I'm trying to think if there's any way to do this with something like a `Promise.race`. This just seems like a flaw in Electron though